gerber/lib/Gerber/Attribute/MD5.hs

```diff
@@ -0,0 +1,30 @@
+{-# language OverloadedStrings #-}
+
+module Gerber.Attribute.MD5
+  ( MD5(..), parseMD5
+  ) where
+
+-- base16-bytestring
+import Data.ByteString.Base16 ( decode )
+
+-- bytestring
+import Data.ByteString ( ByteString )
+import qualified Data.ByteString as ByteString
+
+-- gerber
+import Gerber.Attribute.Attribute ( Field )
+
+-- text
+import Data.Text.Encoding ( encodeUtf8 )
+
+
+newtype MD5 = MD5 ByteString
+  deriving ( Eq, Show )
+
+
+parseMD5 :: MonadFail m => [Field] -> m MD5
+parseMD5 fields = case fields of
+  [field] -> case decode (encodeUtf8 field) of
+    (bytes, "") | ByteString.length bytes == 16 -> pure (MD5 bytes)
+    _ -> fail "Bad .MD5: must consist of exactly 32 hexadecimal digits"
```

Comment:
You aren't actually checking both conditions of this error, which I think might lead to confusion later. I suggest we either do check that the digits are hexadecimal, or just drop this from the error message.

Reply:
I am checking the condition. If they weren't hexadecimal digits then `decode` (from `base16-bytestring`) would fail.

Reply:
Or at least wouldn't return 16 bytes of data with nothing leftover.

Reply:
Ah sorry, I missed the call to decode. Great!